Bottom: 4c45992d2abcdbff9dfe0c32abf6a5967f519528
Top:    9f9e5c599c0a9257cb30f14fbeab3f52529f36b6
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-08-27 11:55:48 +0100

Draw pin / pad names first, so they don't get masked by the pad

Since we use stencil masking to ensure we don't redraw an area
multiple times, confusingly, we need to draw the text first to
ensure it isn't occluded by the pad its-self.

HACK:

We probably draw the names multiple times - as we still use the
common draw.c code which assumes it can draw labels on top of
the pads.


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index caf8223..7a9da64 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1092,6 +1092,13 @@ SetPVColor_inlayer (PinTypePtr Pin, LayerTypePtr Layer, int Type)
 
 
 static int
+pin_name_callback (const BoxType * b, void *cl)
+{
+  DrawPinName ((PinTypePtr) b, 0);
+  return 1;
+}
+
+static int
 pin_inlayer_callback (const BoxType * b, void *cl)
 {
   SetPVColor_inlayer ((PinTypePtr) b, cl, PIN_TYPE);
@@ -1110,6 +1117,7 @@ via_inlayer_callback (const BoxType * b, void *cl)
 static int
 pin_callback (const BoxType * b, void *cl)
 {
+  DrawPinName ((PinTypePtr) b, 0);
   DrawPlainPin ((PinTypePtr) b, false);
   return 1;
 }
@@ -1119,7 +1127,10 @@ pad_callback (const BoxType * b, void *cl)
 {
   PadTypePtr pad = (PadTypePtr) b;
   if (FRONT (pad))
+  {
+    DrawPadName (pad, 0);
     DrawPad (pad, 0);
+  }
   return 1;
 }
 
@@ -1432,6 +1443,9 @@ DrawLayerGroup (int group, const BoxType * screen)
 
       /* Draw pins, vias and pads on this layer */
       if (!global_view_2d && rv) {
+        if (PCB->PinOn &&
+            (group == solder_group || group == component_group))
+          r_search (PCB->Data->pin_tree, screen, NULL, pin_name_callback, Layer);
         if (PCB->PinOn) r_search (PCB->Data->pin_tree, screen, NULL, pin_inlayer_callback, Layer);
         if (PCB->ViaOn) r_search (PCB->Data->via_tree, screen, NULL, via_inlayer_callback, Layer);
         if ((group == component_group && !SWAP_IDENT) ||
